test/graphql.test.ts

```diff
@@ -116,6 +200,54 @@ describe("graphql()", () => {
     });
   });
 
+  it("Variables with TypedDocumentString", () => {
+    const query = new TypedDocumentString<
+      {
+        repository: { issues: { edges: Array<{ node: { title: string } }> } };
+      },
```

Comment:
I don't like that you have to manually type the query. You might as well do `const response = graphql(...) as { repository: { issues: { edges: Array<{ node: { title: string } }> } } }`, this is giving you nothing but false type safety.

Reply:
To clarify: this is a manual annotation that is added as part of the unit tests, in order to test the interface around `TypeDocumentString`, and to avoid running graphql-codegen as part of the test setup 👀 The expectation is fully that users will get `TypeDocumentString` types via graphql-codegen, and both the result and variables will be inferred.